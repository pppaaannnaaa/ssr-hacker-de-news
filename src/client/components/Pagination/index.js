/* eslint-disable jsx-a11y/no-interactive-element-to-noninteractive-role */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { constant } from '../../constant';

const Pagination = (props) => {
  const { pageNumber, isLoading } = props;
  return (
    <div className="carousel-container" style={{ paddingTop: '20px', textAlign: 'right' }}>
      <Link
        to={`/page/${pageNumber - 1}`}
        role="navigation"
        aria-label="Previous Button"
        disabled={pageNumber === 0 || isLoading}
        name={constant.PREV}
        style={{ marginRight: '10px' }}
        type="button"
        className="btn blue darken-4"
      >
        Previous
      </Link>
      <Link
        to={`/page/${pageNumber + 1}`}
        role="navigation"
        aria-label="Next Button"
        disabled={isLoading}
        name={constant.NEXT}
        type="button"
        className="btn blue darken-4"
      >
        Next
      </Link>
    </div>
  );
};

Pagination.propTypes = {
  pageNumber: PropTypes.number,
  isLoading: PropTypes.bool,
};

Pagination.defaultProps = {
  pageNumber: 0,
  isLoading: false,
};

export default Pagination;
