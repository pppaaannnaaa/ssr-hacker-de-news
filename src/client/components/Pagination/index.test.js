import React from 'react';
import { render, cleanup, waitFor, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Pagination from './index';

jest.mock('react-router-dom', () => {
  const originalModule = jest.requireActual('react-router-dom');

  return {
    __esModule: true,
    ...originalModule,
    useParams: jest.fn(() => ({
      id: '1',
    })),
    useHistory: jest.fn(),
    Link: 'Link',
    Route: ({ children, path }) => children({ match: path === '/somewhere' }),
  };
});

describe('<Pagination />', () => {
  let props;

  beforeEach(() => {
    props = {
      pageNumber: 0,
      isLoading: false,
      onClickPrevNext: jest.fn(),
    };
  });

  afterEach(cleanup);

  test('Renders <Pagination /> component correctly', async () => {
    const { container, getByText } = render(<Pagination />);
    expect(container).toMatchSnapshot();

    const nextButton = await waitFor(() => getByText('Next'));
    fireEvent.click(nextButton);

    const prevButton = await waitFor(() => getByText('Previous'));
    fireEvent.click(prevButton);
  });
});
