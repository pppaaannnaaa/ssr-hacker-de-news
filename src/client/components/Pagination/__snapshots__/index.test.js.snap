// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> Renders <Pagination /> component correctly 1`] = `
<div>
  <div
    class="carousel-container"
    style="padding-top: 20px; text-align: right;"
  >
    <link
      aria-label="Previous Button"
      class="btn blue darken-4"
      disabled=""
      name="prev"
      role="navigation"
      style="margin-right: 10px;"
      to="/page/-1"
      type="button"
    >
      Previous
    </link>
    <link
      aria-label="Next Button"
      class="btn blue darken-4"
      name="next"
      role="navigation"
      to="/page/1"
      type="button"
    >
      Next
    </link>
  </div>
</div>
`;
